src/orion/core/worker/experiment_config.py

```diff
@@ -0,0 +1,135 @@
+""" Immutable dataclass containing the experiment configuration. """
+from __future__ import annotations
+
+import typing
+from typing import Any, TypedDict
+
+if typing.TYPE_CHECKING:
+    from orion.core.evc.adapters import CompositeAdapter
+
+
+class EmptyDict(TypedDict):
+    """Empty dictionary."""
+
+
+class DatabaseConfig(TypedDict):
+    """Database configuration."""
+
+    type: str
+
+
+class StorageConfig(TypedDict):
+    """Storage configuration."""
+
+    type: str
+    database: DatabaseConfig
+
+
+class Refers(TypedDict):
+    """Value of the 'refers' property of the Experiment."""
+
+    parent_id: str | None
+    root_id: int | str | None
+    adapter: CompositeAdapter
+
+
+# Note: total=false, because the attribute defaults to {} in Experiment __init__.
+
+
+class RefersConfig(TypedDict, total=False):
+    """Config for the 'refers' attribute of the Experiment."""
+
+    parent_id: str | None
+    root_id: int | str | None
+    adapter: list[dict]
+
+
+class BasicMetaData(TypedDict):
+    """Config for the 'metadata' property of the experiment."""
+
+    user: str
+    """ System user currently owning this running process, the one who invoked **Oríon**. """
+
+    orion_version: str
+    """ Version of **Oríon** which suggested this experiment. `user`'s current **Oríon** version.
+    """
+
+    datetime: datetime
+    """ When was this particular configuration submitted to the database. """
+
+
+class MetaData(BasicMetaData, total=False):
+    """Metadata of the experiment. Also contains some optional keys."""
+
+    # NOTE: Fields that don't appear to be set in the metadata, but that are mentioned in the
```

Comment:
They are set in `resolve_config` when using the commandline interface.

Reply:
Okay, good to know. They are still optional, from a typing perspective.